slack_done.gs

```diff
@@ -102,26 +107,21 @@ function done_process_modal(args){
   var access_token = PropertiesService.getScriptProperties().getProperty('ACCESS_TOKEN'); // confidential Slack API access token
 
   // process done command
-  var out_message = done(args);
+  // Note: done returns JSON.stringify({blocks:[...]})
+  // We need to unpack and repack to mark it as ephemeral.
+  var payload = JSON.parse(done(args));
+  payload["response_type"] = "ephemeral";
+  payload = JSON.stringify(payload);
 
   // send reply to slack user with response_url functionality
   var response_url = args.response_url;
-  var options = {
-    method: "post",
-    contentType: 'application/json; charset=utf-8',
-    headers: {Authorization: 'Bearer ' + access_token},
-    payload: JSON.stringify({
-      "text": out_message,
-      "response_type": "ephemeral"
-    })
-  };
-  var return_message = UrlFetchApp.fetch(response_url, options).getContentText(); // Send post request to Slack response_url
+  var return_message = postToSlack(payload, response_url)
 
   // update log sheet
   var log_sheet = new LogSheetWrapper();
   log_sheet.appendRow([new Date(), args.uniqueid, 'admin','confirmDoneUser', return_message]);
 
-  return ""; // Return empty string to close modal
+  return textToJsonBlocks("Thanks!");
```

Comment:
does this work? I think the string needs to be empty otherwise the modal won't close

Reply:
Good point - didn't test this path. Reverted.